src/intersection-observer.ts

```diff
@@ -202,17 +214,22 @@ export function generateEntry(
   let width = Math.min(rootBounds.left + rootBounds.width, clientRect.right) - intersectX;
   let height = Math.min(rootBounds.top + rootBounds.height, clientRect.bottom) - intersectY;
 
-  let intersectionRect: ClientRect = {
-    left: width >= 0 ? intersectX : 0,
-    top: intersectY >= 0 ? intersectY : 0,
+  const interLeft = width >= 0 ? intersectX : 0;
+  const interTop = intersectY >= 0 ? intersectY : 0;
+  let intersectionRect: ClientRect & SpanielRect = {
+    left: interLeft,
+    top: interTop,
+    x: interLeft,
+    y: interTop,
```

Comment:
Are these new? I might have missed where are `x` and `y` read?

Reply:
no, it's just that we previously were not actually returning a https://developer.mozilla.org/en-US/docs/Web/API/DOMRect
so adding it to be closer to the native behavior.

Reply:
Looks like `ClientRect` is a subset of `SpanielRect` so there's no need to do `ClientRect & SpanielRect`.  I also noticed there's built-in `DOMRectReadOnly ` from https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts. So https://github.com/linkedin/spaniel/blob/6b74e4d523a089d128ef88422fa3c13320e50afe/src/interfaces.ts#L83 and https://github.com/linkedin/spaniel/pull/143#discussion_r653814703 might not be needed

Reply:
Ok I created `export type DOMRectPojo = Omit<DOMRectReadOnly, 'toJSON'>;`
Instead of `ClientRect & SpanielRect`